client/state/data-layer/wpcom/sites/media/index.js

```diff
@@ -59,6 +60,7 @@ export const requestMediaSuccess =
 	( { siteId, query }, data ) =>
 	( dispatch, getState ) => {
 		if (
+			getCurrentRoute( getState() )?.startsWith( '/media/' ) &&
 			! isEqual(
 				omit( query, 'page_handle' ),
 				omit( getNextPageQuery( getState(), siteId ), 'page_handle' )
```

Comment:
@flootr Tagging you here since you originally make this fix for https://github.com/Automattic/wp-calypso/pull/44579

The query check was preventing the request from the `QueryMedia` react component from completing anywhere except the Media Library in calypso because the "next page query" url always had `number` and none of the passed in query filters (like `mime_type`) as properties and the custom query that I provided to the component just had my `mime_type` property. As a result it would always fall into this condition.

It doesn't look like this change I'm making is causing the original issue to come back, unless I'm not testing correctly. Just hoping to get a second set of eyes on this incase you see an issue with the change.

Alternatively, it sounds like the original issue that was being fixed has something to do with old requests not being cancelled... I wonder if there is another way to solve for that that wouldn't require a special case to skip this check for the `/media` route? Something like storing a "current query" instead and just comparing that with whatever is coming back in the result?

Reply:
I've actually replaced this change with a bug fix. Took a bit of investigating, but it looks like the `<QueryMedia>` component wasn't setting the `query` state, which was causing the `isEqual` check in the above code to trigger when it shouldn't.

This has now been fixed directly in the `<QueryMedia>` component!